test/unit/navigation/NavigatorSpec.scala

```diff
@@ -80,13 +89,209 @@ class NavigatorSpec extends SpecBase {
+          routes.CommodityCodeRulingReferenceController.onPageLoad(NormalMode)
+      }
+
+      "redirect to RegisteredAddressForEoriPage from CommodityCodeRulingReferencePage" in {
+        navigator.nextPage(CommodityCodeRulingReferencePage, NormalMode)(mock[UserAnswers]) shouldBe
+          routes.RegisteredAddressForEoriController.onPageLoad(NormalMode)
+      }
+
+      "redirect to RegisteredAddressForEoriPage when user selects NO from SimilarItemCommodityCodePage" in {
+        val mockUserAnswers = mock[UserAnswers]
+
+        when(mockUserAnswers.get(SimilarItemCommodityCodePage)).thenReturn(Some(false))
+
+        navigator.nextPage(SimilarItemCommodityCodePage, NormalMode)(mockUserAnswers) shouldBe
+          routes.RegisteredAddressForEoriController.onPageLoad(NormalMode)
+      }
+
+      "redirect to EnterContactDetailsPage from RegisteredAddressForEoriPage" in {
+        navigator.nextPage(RegisteredAddressForEoriPage, NormalMode)(mock[UserAnswers]) shouldBe
+          routes.EnterContactDetailsController.onPageLoad(NormalMode)
+      }
+
+      "redirect to CheckYourAnswersPage from EnterContactDetailsPage" in {
+        navigator.nextPage(EnterContactDetailsPage, NormalMode)(mock[UserAnswers]) shouldBe
+          routes.CheckYourAnswersController.onPageLoad()
+      }
+
+      "redirect to DeclarationPage from CheckYourAnswersPage" in {
+        navigator.nextPage(CheckYourAnswersPage, NormalMode)(mock[UserAnswers]) shouldBe
+          routes.DeclarationController.onPageLoad(NormalMode)
+      }
+
+      "redirect to ConfirmationPage from DeclarationPage" in {
+        navigator.nextPage(DeclarationPage, NormalMode)(mock[UserAnswers]) shouldBe
+          routes.ConfirmationController.onPageLoad()
+      }
+
+      "redirect to self if no answers were entered" in {
+        val mockUserAnswers = mock[UserAnswers]
+
+        when(mockUserAnswers.get(any())(any())).thenReturn(None)
+
+        navigator.nextPage(AddConfidentialInformationPage, NormalMode)(mockUserAnswers) shouldBe
+          routes.AddConfidentialInformationController.onPageLoad(NormalMode)
+
+        navigator.nextPage(WhenToSendSamplePage, NormalMode)(mockUserAnswers) shouldBe
+          routes.WhenToSendSampleController.onPageLoad(NormalMode)
+
+        navigator.nextPage(CommodityCodeBestMatchPage, NormalMode)(mockUserAnswers) shouldBe
+          routes.CommodityCodeBestMatchController.onPageLoad(NormalMode)
+
+        navigator.nextPage(LegalChallengePage, NormalMode)(mockUserAnswers) shouldBe
+          routes.LegalChallengeController.onPageLoad(NormalMode)
+
+        navigator.nextPage(SelectApplicationTypePage, NormalMode)(mockUserAnswers) shouldBe
+          routes.SelectApplicationTypeController.onPageLoad(NormalMode)
+
+        navigator.nextPage(SimilarItemCommodityCodePage, NormalMode)(mockUserAnswers) shouldBe
+          routes.SimilarItemCommodityCodeController.onPageLoad(NormalMode)
+      }
     }
 
     "in Check mode" must {
```

Comment:
Is it worthwhile adding tests here for the yes/no journeys in checkmode?

Reply:
Yes in fact that was the main thing I was trying to achieve today before I got sidetracked 😂